lib/ansible/errors/__init__.py

```diff
@@ -334,3 +334,8 @@ class AnsiblePluginCircularRedirect(AnsiblePluginError):
 class AnsibleCollectionUnsupportedVersionError(AnsiblePluginError):
     '''a collection is not supported by this version of Ansible'''
     pass
+
+
+class AnsibleFilterTypeError(AnsibleTemplateError, TypeError):
```

Comment:
Overall, I think this change fixes the problem well. I'm just not sure about the pros/cons of having a custom exception inherit from multiple exceptions, one custom and one built it. It may be ok, I'm just not sure.

Reply:
```
BaseException
  +-- Exception
  |     +-- TypeError
              +-- AnsibleFilterTypeError
  +-- AnsibleError
        +-- AnsibleRuntimeError
              +-- AnsibleTemplateError
                    +-- AnsibleFilterTypeError
```

Reply:
its only way to thread needle between us capturing typeerror and reusing on the 'undefined case' and still passing through as a filter error whichalso has it's own special handling.